staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -303,7 +305,13 @@ func NewSharedIndexInformer(lw ListerWatcher, exampleObject runtime.Object, defa
 func NewSharedIndexInformerWithOptions(lw ListerWatcher, exampleObject runtime.Object, options SharedIndexInformerOptions) SharedIndexInformer {
 	realClock := &clock.RealClock{}
 
+	informerName := naming.GetNameFromCallsite(internalPackages...)
+	typeName := reflect.TypeOf(exampleObject).Elem().Name()
+	metricName := makeValidPromethusMetricName(fmt.Sprintf("informer_%s_type_%s", informerName, typeName))
+
```

Comment:
The name of this informer metric is not ideal. How can we define a more readable and maintainable informer metric name?
PS: I don't have a good idea at the moment.

Reply:
@wojtek-t Please help take a look at this.

Reply:
Sorry, I'm on PTO pretty soon so I didn't have time to review this PR properly.

Let's please make sure that we don't end up with duplicate metric names. E.g. in controller-runtime we could create multiple informers for the same GVK (one per Unstructured, PartialObjectMeta, regular Go Type).

Reply:
Thank you @sbueringer. I'll modify the code to use `reflect.TypeOf(exampleObject).Elem().String()` instead of `reflect.TypeOf(exampleObject).Elem().Name()`.

This change will result in metric names like `informer_somecontroller_type_k8s.io_api_core_v1_Pod` instead of `informer_somecontroller_type_Pod`, making collisions less likely, especially when dealing with types like Unstructured or PartialObjectMetadata alongside specific Go types for the same GVK.

Or do you have any good suggestions?

Reply:
Sounds like this should work for CR. Not sure if there are cases where folks would basically create entirely duplicate informers and then end up with problems because the same metric will be registered twice.

I'm on PTO starting today, so I can't follow-up. But maybe @alvaroaleman can chime. in.